src/EnergyPlus/PluginManager.cc

```diff
@@ -484,8 +484,23 @@ PluginManager::PluginManager(EnergyPlusData &state) : eplusRunningViaPythonAPI(s
 
     initPython(state, pathToPythonPackages);
 
-    // Take control of the global interpreter lock while we are here, make sure to release it...
-    PyGILState_STATE gil = PyGILState_Ensure();
+    // GilGrabber is an RAII helper that will ensure we release the GIL (including if we end up throwing)
+    struct GilGrabber
+    {
+        GilGrabber()
+        {
+            gil = PyGILState_Ensure();
+        }
+        ~GilGrabber()
+        {
+            PyGILState_Release(gil);
+        }
+
+        PyGILState_STATE gil;
+    };
+
+    // Take control of the global interpreter lock
+    GilGrabber gil_grabber;
```

Comment:
Ok, so the issue was that we were NOT releasing the Python Global Interpreter Lock after acquiring it. (I tested a C version of the same test, which was correctly throwing and exiting as expected, so I figured it was the GIL)


Initially in 552f5ddfa923d85d33dbef9835a411c3478f445e I put a try / catch around plugin.setup()

```diff
    for (auto &plugin : state.dataPluginManager->plugins) {
+        try {
            plugin.setup(state);
+        } catch (const FatalError &e) {
+            PyGILState_Release(gil);
+            throw e;
 +       }
    }
```

But I realized it was probably a better idea to just use RAII, so that there's no chance we forget to release the Python GIL